### PR TITLE
Add support for aeson 2.2

### PR DESCRIPTION
**Pull request by Tristano8**
Do not merge until `attoparsec-aeson` has a compat package on Hackage

### CODE REVIEW

**Comment by endgame** [comment]
https://github.com/haskell/aeson/issues/1048 has been completed and the compatibility package released.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog][chg] and this project adheres to
 [Haskell's Package Versioning Policy][pvp]
 
+## unreleased - 2023-07-06
+
+  - Add support for aeson 2.3
```
Comment (IamfromSpace) [inline CHANGELOG.md:10]: This should be 2.2, rather than 2.3.

Comment (Tristano8) [reply CHANGELOG.md]: I've updated the commit. Thanks

**Comment by IamfromSpace** [comment]
Alrighty, attoparsec-aeson-2.1 is now in stackage, so I think this is now through the last hurdle.  Since it's only in stackage nightly and lts-21, we're gonna see failures for all stack builds other than nightly.

In a separate PR I'll address the builds and then cut a new release.  Getting stackage nightly compatibility will be great.

**Pull request merged**